rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -395,18 +395,29 @@ const rosbag2_cpp::Writer & RecorderImpl::get_writer_handle()
 
 void RecorderImpl::pause()
 {
-  paused_.store(true);
-  RCLCPP_INFO_STREAM(node->get_logger(), "Pausing recording.");
+  std::lock_guard<std::recursive_mutex> lock(state_transaction_mutex_);
+  if (paused_ == true) {
+    RCLCPP_DEBUG(node->get_logger(), "Recording has already paused.");
+  } else {
+    paused_.store(true);
+    RCLCPP_INFO_STREAM(node->get_logger(), "Pausing recording.");
+  }
 }
 
 void RecorderImpl::resume()
```

Comment:
this can be implemented as `pause(false)` to avoid the implementation redundancy. any thoughts?

Reply:
@fujitatomoya Well. From one hand yes it could be simplified with `pause(false)` but we deliberately choose to use `pause()` and `resume()` methods to have a clean and more obvious API for managing rosbag2 recorder state from remote via service calls and during distributed recording.  
This is to facilitate `Pause`, `Resume` and `IsPaused` service calls. 
In addition IMO `pause(false)` is not very clear what does it really mean those `false`.